tests/test_RBAC.py

```diff
@@ -58,6 +56,44 @@ def test_encrypt_request_payload(consumer_wallet, publisher_wallet, monkeypatch)
         "type": "address",
         "value": publisher_wallet.address,
     }
+    assert payload["data"] == [document]
+
+    ddo = copy.deepcopy(json_dict)
+    req["data"] = ddo
+
+    validator = RBACValidator(request_name="EncryptRequest", request=req)
+    payload = validator.build_payload()
+    assert payload["data"] == ddo
+
+
+@pytest.mark.unit
+def test_wrong_encrypt_request_payload(consumer_wallet, publisher_wallet, monkeypatch):
```

Comment:
please add tests for the individual functions as well, the test coverage is very poor on your added code. How do I know this? The imbrications many and hard to read. Poorly readable means poorly testable, so that's why you couldn't add tests.